### PR TITLE
(#165) Rename Verifies to Satisfies and improve API

### PR DESCRIPTION
**Pull request by victornoel**
This is a followup on #165 to improve Verifies.
- It was renamed to Satisfies to match the naming using in other similar tooling such as assertj.
- The API was improved to support any type of feature extraction

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat assign @andreoss

**Comment by 0crat** [comment]
Job [#242](https://github.com/llorllale/cactoos-matchers/issues/242) is already in scope

File: src/main/java/org/llorllale/cactoos/matchers/Satisfies.java [hunk 1]
```diff
@@ -27,24 +27,61 @@
...
+    public <U> Satisfies(
+        final Matcher<? super U> matcher, final String fdesc,
+        final Func<? super T, ? extends U> extractor
+    ) {
         super(
             new MatcherOf<>(
-                func,
-                desc -> desc.appendText("verifies"),
-                (obj, desc) -> desc.appendText("was ").appendValue(obj)
+                obj -> matcher.matches(extractor.apply(obj)),
```
Comment (andreoss) [inline src/main/java/org/llorllale/cactoos/matchers/Satisfies.java:73]: @victornoel Besides the bug-prone double calling of 'apply', it looks fine.

**Review by andreoss — APPROVED** [review]

File: src/main/java/org/llorllale/cactoos/matchers/Satisfies.java (hunk 1 above)
Comment (victornoel) [reply src/main/java/org/llorllale/cactoos/matchers/Satisfies.java:73]: @andreoss yep, we can discuss this double-calling in #239 to see if we can find some idea on how to improve this

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 3 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**